### PR TITLE
A40 MRG fixes

### PR DESCRIPTION
**Pull request by xpander53**
Signed-off-by: Joel Maner <xpander53@yahoo.com>

### CODE REVIEW

**Review by lisadier — APPROVED** [review]
Looks great!

File: src/main/resources/hl7/resource/PatientMRG.yml [hunk 1]
```diff
@@ -36,17 +36,35 @@ active:
...
-           specs: $Patient
+      type:
+        type: STRING
+        valueOf: 'replaced-by'
+      other:
+        required: true
+        valueOf: datatype/Reference
+        expressionType: resource
+        specs: $Patient
+link_2:
```
Comment (klwhaley) [inline src/main/resources/hl7/resource/PatientMRG.yml:54]: can we add a comment that link_2 is due to different structure of segment groups in A40, so we remember later why we did this?

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java [hunk 2]
```diff
@@ -159,10 +158,9 @@ public void validateHappyPathADT_A34WithMRG() {
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java:158]: Do we have an ADT_A34 test?

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java [hunk 3]
```diff
@@ -253,6 +250,110 @@ public void validateTwoMRGs() {
...
+
+       // Verify the patient four identifier value
+       assertThat(ResourceUtils.getValueAsString(patientFourIdentifierList.get(0), "value")).isEqualTo("MR4");
+       // Verify the patient four identifier value is marked as old
+       assertThat(ResourceUtils.getValueAsString(patientFourIdentifierList.get(0), "use")).isEqualTo("Enumeration[old]");
+
+       //The fourth patient should NOT be active.
+       assertThat(ResourceUtils.getValueAsString(patientFour, "active")).isEqualTo("BooleanType[false]");
+
+       // We should have link.other.reference to the PID (3r) patient
```
Comment (klwhaley) [inline src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java:351]: nit 3r? 3rd?

Comment (xpander53) [reply src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java:351]: Fixed

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java (hunk 2 above)
Comment (xpander53) [reply src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java:158]: Yes it's the first test in the suite

File: src/main/resources/hl7/resource/PatientMRG.yml (hunk 1 above)
Comment (xpander53) [reply src/main/resources/hl7/resource/PatientMRG.yml:54]: Will do

Comment (LisaWellman) [inline src/main/resources/hl7/resource/PatientMRG.yml:54]: looks like this is the same as the previous, can it not be combined (would that be a 'complex condition')?

File: src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java (hunk 2 above)
Comment (klwhaley) [reply src/test/java/io/github/linuxforhealth/hl7/segments/HL7MergeFHIRConversionTest.java:158]: just missed it, closing

File: src/main/resources/hl7/resource/PatientMRG.yml (hunk 1 above)
Comment (LisaWellman) [reply src/main/resources/hl7/resource/PatientMRG.yml:54]: It is not clear to me why we need the condition check -- when would we NOT want these to fire?

**Review by klwhaley — APPROVED** [review]
approving with a few minor comments

File: src/main/resources/hl7/resource/PatientMRG.yml (hunk 1 above)
Comment (xpander53) [reply src/main/resources/hl7/resource/PatientMRG.yml:54]: I added a comment to explain.
It's triggering different "link"s depending on the event trigger.
A40 has patient in a group and A34 doesn't.

**Pull request merged**